### PR TITLE
Add SmartDec SAST tool

### PR DESCRIPTION
**Pull request by ATSDec**
Add information about **SmartDec** SAST tool to **Static analysis** section

### CODE REVIEW

**Comment by jishanshaikh4** [comment]
@ATSDec Seems like a "neutral" point to include here. But it just didn't add more insightful results to the learning and research towards modern CPP.

**Comment by ATSDec** [comment]
SmartDec Scanner supports all modern versions of languages, up to C++ 17/20 and all modern build systems. In our tool, much attention is paid to vulnerabilities and security inaccuracy. SmartDec Scanner distinctive feature is the ability to analyze not only source code, but also executable files (i.e. binaries, .exe and .dll files for C++), our tool can do it without debug info.  We think that it is important that in this list people should have a large choice of analyzers that are capable to provide high-quality results.

**Comment by rigtorp** [comment]
Doesn't have anything to do with learning modern C++.

**Comment by ATSDec** [comment]
@rigtorp Thank you for the review.

Although initially the PR was rejected, currently there is no clear point on why it has been determined that SmartDec Scanner doesn't contribute into modern C++ ecosystem. Thus we propose reopening the issue for gathering more discussion details into it and ensuring that the list is not being personally biased.

As could be noticed, the list currently includes other notable C++ static analysis tools, in particular, clang-analyzer and cppcheck. One of the jobs SmartDec Scanner automates is an effortless running of enhanced versions of these utilities on supplied code base and, afterwards, cautious and thorough verification of produced results, filtering out most of false positives. Therefore, the contribution of the SmartDec Scanner to learning and leveraging modern C++ is no less than of the tools already included into the list.

Moreover, SmartDec Scanner does run a vast variety of checks, automating vulnerability discovery as well as detecting generic errors and discrepancies with recommended code writing practices. For instance, C++ engine is able detect safety, performance, thread safety and portability issues and to check the code base against best practices guidelines, such as [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines (https://github.com/isocpp/CppCoreGuidelines)). Such checks obviously does help C++ developers better understand language semantics as well as properties of their code base and ensure guideline acceptance for it.

Please let us know if you need any further details.

Not to mention, that the Best Practices section is missing static analysis step which is crucial for modern [Secure SDLC](https://www.owasp.org/index.php/OWASP_Secure_Software_Development_Lifecycle_Project (https://www.owasp.org/index.php/OWASP_Secure_Software_Development_Lifecycle_Project)). Not including static analysis into suggested best practices would be simply misleading, as any modern development workflow aimed for delivering quality production-ready products involves static analysis step(s). Thus, providing users of the list with as many alternatives for performing static analysis on modern C++ code bases as possible should be the end goal, as such variety enables list users to leverage their development workflow automation for writing more robust code.

Summing up, we would be grateful if you could elaborate more on the acceptance criteria for program analysis products and tooling in general. This would also help future contributors understand requirements for getting into the list, thus making the C++ community a greater place 👍

**Comment by rigtorp** [comment]
The link leads to some company's marketing page. It's useless. I don't even have Coverity and PVS studio which are well known and highly regarded tools.

Yes best practices section needs work.

This is just my hobby, I put interesting and useful stuff here.
